test/test-execute/exec-temporaryfilesystem-options.service

```diff
@@ -4,8 +4,14 @@ Description=Test for TemporaryFileSystem with mount options
 [Service]
 Type=oneshot
 
+# The mount options default to "mode=0755,nodev,strictatime".
+# Let's override some of them, and test the behaviour of "ro".
+TemporaryFileSystem=/var:ro,mode=0700,nostrictatime
+
 # Check /proc/self/mountinfo
-ExecStart=/bin/sh -c 'test $$(awk \'$$5 == "/var" { print $$6 }\' /proc/self/mountinfo) = "ro,nodev,relatime"'
-ExecStart=/bin/sh -c 'test $$(awk \'$$5 == "/var" { print $$11 }\' /proc/self/mountinfo) = "ro,mode=700"'
+ExecStart=/bin/sh -x -c 'test "$$(awk \'$$5 == "/var" && $$11 !~ /(^|,)ro(,|$)/ { print $$6 }\' /proc/self/mountinfo)" = ""'
+ExecStart=/bin/sh -x -c 'test "$$(awk \'$$5 == "/var" && $$11 !~ /(^|,)mode=700(,|$)/ { print $$6 }\' /proc/self/mountinfo)" = ""'
 
-TemporaryFileSystem=/var:ro,mode=0700,nostrictatime
+ExecStart=/bin/sh -x -c 'test "$$(awk \'$$5 == "/var" && $$6 !~ /(^|,)ro(,|$)/ { print $$6 }\' /proc/self/mountinfo)" = ""'
+ExecStart=/bin/sh -x -c 'test "$$(awk \'$$5 == "/var" && $$6 !~ /(^|,)nodev(,|$)/ { print $$6 }\' /proc/self/mountinfo)" = ""'
+ExecStart=/bin/sh -x -c 'test "$$(awk \'$$5 == "/var" && $$6 ~ /(^|,)strictatime(,|$)/ { print $$6 }\' /proc/self/mountinfo)" = ""'
```

Comment:
I think it would be great if `/bin/sh` and `test` weren't run. I believe that it is possible to write a script in `awk` that would do everything we need. Although it would require more effort to make it work, I think it is worth it. What do you think?

Reply:
I think at least `/bin/sh` should be removed because the quoted quotes are a bit too much :-)

Reply:
Yeah. But this is just a test. If it works, then I think it is OK. Moreover, I am not good at writing code in awk... Do you have any suggestion?

Reply:
Yeah, the nested single quotes takes this into horrifying territory for me.  And there's some `$` in the regex that I forgot to double up, who knows what will happen there.

I can draft a cleaner script... but then it feels like it should be a separate file, to avoid escaping lots of newlines and still having to double up the `$`.  The problem is the test service doesn't know where to find my nice script.  I tried using `sh -c "$$SYSTEMD_TEST_DATA/exec-temporary-filesystem-options.sh"` but it didn't work; the env var was empty.

```
$5 == "/var" {
        if ($11 !~ /(^|,)ro(,|$)/ ||
            $11 !~ /(^|,)mode=700(,|$)/ ) {
                print "Unexpected fs options: "
                print $11
                exit 1
        }
        if ($6 !~ /(^|,)ro(,|$)/ ||
            $6 !~ /(^|,)nodev(,|$)/ ||
            $6 ~ /(^|,)strictatime(,|$)/ ) {
                print "Unexpected vfs options: "
                print $6
                exit 1
        }
}
```

Reply:
@yuwata, the script I mentioned in https://github.com/systemd/systemd/pull/7908#discussion_r169618475 works too, but I doubt that anyone would be glad to see it, especially when it started to fail because of a regression or anything like this :-) Basically, I think that the tests should be easy to understand and currently it is relatively hard to grasp that the first two line just check that `ro` and `mode=700` are somewhere in the 11th field.

@sourcejedi, there is the function `get_testdata_dir` that should return something useful when `$SYSTEMD_TEST_DATA` is not set, but I think that `\` in the unit file would be fine too.

I haven't written code in AWK for a long time, but I remember that there are hashes and `split` in AWK. I wonder if they could be used to get rid of the regular expressions.

Reply:
I've just skimmed over `man awk` and I have to admit that hashes and `split` are overkill :-)